src/transformers/models/owlvit/modeling_owlvit.py

```diff
@@ -1301,12 +1309,84 @@ def image_text_embedder(
 
         return (text_embeds, image_embeds, text_model_last_hidden_state, vision_model_last_hidden_state)
 
+    def embed_image_query(self, query_pixel_values: torch.FloatTensor) -> torch.FloatTensor:
+        (_, class_embeds) = self.class_predictor(query_pixel_values)
+        mean_embeds = torch.mean(class_embeds, axis=1)
+        mean_sim = torch.einsum("bd,bid->bi", mean_embeds, class_embeds)
+        idxs = torch.argmin(mean_sim, axis=1)
+        query_embeds = torch.stack([class_embeds[i][idxs[i]] for i in range(idxs.size()[0])])
+        return query_embeds
```

Comment:
This seems mostly correct to me but it seems to be missing the high predicted box overlap requirement.

We want to select the class_embedding that:
- Has the highest average dissimilarity when compared to other class embeddings. This means we need to compute the dissimilarity of each entry of class_embeds to all other entries, and take their average:
<img width="223" alt="Screenshot 2022-09-07 at 22 31 36" src="https://user-images.githubusercontent.com/8944735/188961654-2fa14bb1-a5ce-4f1e-8a74-0abec6e6195c.png">
- Has a corresponding predicted box that has at least 0.65 overlap with the query image

You would need to call self.box_predictor() to check for the second requirements. If the second condition is not satisfied, we can return None

Reply:
Ah yes I missed this, will add